crates/oxc_linter/src/rules/eslint/no_extra_label.rs

```diff
@@ -0,0 +1,295 @@
+use oxc_ast::{ast::LabelIdentifier, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_extra_label_diagnostic(label: &LabelIdentifier) -> OxcDiagnostic {
+    let label_name = &label.name;
+    OxcDiagnostic::warn(format!("This label '{label_name}' is unnecessary"))
+        .with_help(format!("Remove this label. It will have the same result because the labeled statement '{label_name}' has no nested loops or switches",))
+        .with_label(label.span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoExtraLabel;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow unnecessary labels.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// If a loop contains no nested loops or switches, labeling the loop is unnecessary.
+    /// ```js
+    /// A: while (a) {
+    ///     break A;
+    /// }
+    /// ```
+    /// You can achieve the same result by removing the label and using `break` or `continue` without a label.
+    /// Probably those labels would confuse developers because they expect labels to jump to further.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// A: while (a) {
+    ///     break A;
+    /// }
+    ///
+    /// B: for (let i = 0; i < 10; ++i) {
+    ///     break B;
+    /// }
+    ///
+    /// C: switch (a) {
+    ///     case 0:
+    ///         break C;
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// while (a) {
+    ///     break;
+    /// }
+    ///
+    /// for (let i = 0; i < 10; ++i) {
+    ///     break;
+    /// }
+    ///
+    /// switch (a) {
+    ///     case 0:
+    ///         break;
+    /// }
+    ///
+    /// A: {
+    ///     break A;
+    /// }
+    ///
+    /// B: while (a) {
+    ///     while (b) {
+    ///         break B;
+    ///     }
+    /// }
+    ///
+    /// C: switch (a) {
+    ///     case 0:
+    ///         while (b) {
+    ///             break C;
+    ///         }
+    ///         break;
+    /// }
+    /// ```
+    NoExtraLabel,
+    style,
+    fix
+);
+
+impl Rule for NoExtraLabel {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::BreakStatement(break_stmt) = node.kind() {
+            if let Some(label) = &break_stmt.label {
+                report_label_if_extra(label, node, LabelUsage::Break, ctx);
+            }
+        }
+        if let AstKind::ContinueStatement(cont_stmt) = node.kind() {
+            if let Some(label) = &cont_stmt.label {
+                report_label_if_extra(label, node, LabelUsage::Continue, ctx);
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+enum LabelUsage {
```

Comment:
we can drop this enum. we pass in `node` into `report_label_if_extra` so we already know whether it is a break/continue stmt

https://github.com/oxc-project/oxc/pull/8181/commits/c8ae89b2523c71626201861a0611f36ae78f9b3c